stingray/lightcurve.py

```diff
@@ -583,70 +583,80 @@ def join(self, other):
+                new_counts_err = np.zeros_like(new_counts)
 
-            try:
-                count2 = other.counts[np.where(other.time == time)[0][0]]
-                count2_err = other.counts_err[np.where(other.time == time)[0][0]]
-            except IndexError:
-                count2 = None
-                count2_err = None
-
-            if count1 is not None:
-                if count2 is not None:
-                    # Average the overlapping counts
-                    new_counts.append((count1 + count2) / 2)
-
-                    if self.err_dist.lower() != other.err_dist.lower():
-                        simon("Lightcurves have different statistics!"
-                              "We are setting the errors to zero.")
-                        new_counts_err = np.zeros_like(new_counts)
-                    elif self.err_dist.lower() in valid_statistics:
-                        new_counts_err.append(np.sqrt(((count1_err**2) +
-                                                      (count2_err**2)) / 2))
-                    # More conditions can be implemented for other statistics
-                    else:
-                        raise StingrayError("Statistics not recognized."
-                                            " Please use one of these: "
-                                            "{}".format(valid_statistics))
-                else:
-                    new_counts.append(count1)
-                    new_counts_err.append(count1_err)
+            elif self.err_dist.lower() in valid_statistics:
+                valid_err = True
+            # More conditions can be implemented for other statistics 
             else:
-                new_counts.append(count2)
-                new_counts_err.append(count2_err)
+                raise StingrayError("Statistics not recognized."
+                                    " Please use one of these: "
+                                    "{}".format(valid_statistics))
+
+
+            from collections import Counter
+            counts = Counter()
+            counts_err = Counter()
+
+            for i, time in enumerate(first_lc.time):
+                counts[time] = first_lc.counts[i]
+                counts_err[time] = first_lc.counts_err[i]
 
+            for i, time in enumerate(second_lc.time):
+            
+                if (counts[time] != 0): #Common time
+
+                    counts[time] = (counts[time] + second_lc.counts[i]) / 2  #avg
+                    counts_err[time] = np.sqrt(( ((counts_err[time]**2) + (second_lc.counts_err[i] **2)) / 2))
+
+                else:
+                    counts[time] = second_lc.counts[i]
+                    counts_err[time] = second_lc.counts_err[i]
+
+            new_time = list(counts.keys())
+            new_counts = list(counts.values())
+            if(valid_err):
+                new_counts_err = list(counts_err.values())
+            
+            del[counts]
```

Comment:
I know what you are trying to do, but I'm not sure that you're actually deleting it.
I would expect it to be `del counts`, not `del[counts]`. Have you tried using `counts` after, to verify that it's actually deleted?